
2019-01-31 Yiqiao Jin <ahren2040@g.ucla.edu>
* timespec.h(timespec_cmp):Change function to comparison of time to avoid integer overflow


--- timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespec_patch.h	2019-01-31 13:27:15.148371000 -0800
@@ -45,8 +45,9 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if(a.tv_nsec == b.tv_nsec)
+    return 0;
+  return a.tv_nsec < b.tv_nsec ? -1:1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
